app/controllers/mixins/generic_button_mixin.rb

```diff
@@ -0,0 +1,48 @@
+module Mixins
+  module GenericButtonMixin
+    # handle buttons pressed on the button bar
+    def button
+      @edit = session[:edit]                          # Restore @edit for adv search box
+      params[:display] = @display if ["images", "instances"].include?(@display)  # Were we displaying vms/hosts/storages
+      params[:page] = @current_page unless @current_page.nil?   # Save current page for list refresh
+
+      if params[:pressed].starts_with?("image_", # Handle buttons from sub-items screen
+                                       "instance_")
+
+        pfx = pfx_for_vm_button_pressed(params[:pressed])
+        process_vm_buttons(pfx)
+
+        # Control transferred to another screen, so return
```

Comment:
indeed, it's for separate PR to clean it up, this was basically copied to all cloud controllers